docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,254 @@
+
+- **Introduce the `garbage-collect` Command to `etcdbrctl`:**
+
+  - `etcd-backup-restore` will be enhanced with a new command `garbage-collect`, which:
+    - Performs garbage collection of snapshots in the object store according to `etcd-backup-restore`'s [garbage collection policy](https://github.com/gardener/etcd-backup-restore/blob/master/docs/usage/garbage_collection.md#gc-policies).
+
+  This functionality is needed to remove old snapshots whose immutability has expired, preventing storage from growing indefinitely.
+
+- **Update the `Etcd` CRD:**
+
+  - Add a new field `immutability.retentionType` under `etcd.spec.backup.store` to specify the type of immutability config.
+
+- **etcd Controller Logic:**
+
+  - When hibernation is requested:
+    - The controller removes the etcd client ports `2380` and `2379` from the `etcd-client` service, leaving only the `etcd-backup-restore` port `8080`, effectively stopping etcd client traffic.
+    - The controller creates an `EtcdOperatorTask` to trigger an on-demand full snapshot.
+    - The controller scales down the etcd cluster by setting `StatefulSet.spec.replicas` to zero.
+    - The controller periodically creates the `ExtendEtcdSnapshotImmutabilityTask` if `etcd.spec.backup.store.immutability.retentionType` is set to `"Bucket"` and based on `etcd.spec.backup.fullSnapshotSchedule`.
+
+- **`ExtendEtcdSnapshotImmutabilityTask` Specification:**
+
+  - Runs `etcdbrctl renew-snapshot` to extend the immutability of the latest snapshot.
+  - Runs `etcdbrctl garbage-collect --garbage-collection-policy <policy>` to remove old snapshots.
+
+- **EtcdOperatorTask Controller Logic:**
+
+  - The operator-tasks controller reacts to the creation of the custom resource and deploys a job named `<etcd-name>-extend-immutability`, which is the `ExtendEtcdSnapshotImmutabilityTask` job.
+  - The controller reports metrics regarding the `ExtendEtcdSnapshotImmutabilityTask` job, which can be used to raise alerts for operators if immutability has not been extended.
+
+By periodically re-uploading the latest snapshot during hibernation, the authors ensure that the immutability period is extended, and the backups remain **protected throughout the hibernation period**.
+
+## Compatibility
+
+The proposed changes are fully compatible with existing etcd clusters and backup processes.
+
+- **Backward Compatibility:**
+  - Existing clusters without immutable buckets will continue to function without change.
+  - The introduction of the `ExtendEtcdSnapshotImmutabilityTask` does not affect clusters that are not hibernated.
+- **Forward Compatibility:**
+  - Clusters can opt-in to use immutable backups by configuring the bucket accordingly.
+  - The controller's logic to handle hibernation is additive and does not interfere with existing workflows.
+
+## Risks and Mitigations
+
+- **Increased Storage Costs:**
+  - **Risk:** Copying snapshots or frequent snapshots may lead to increased storage usage.
```

Comment:
There are 2 things that you mention here -> `copying snapshots` and `frequent snapshots`. The latter is not part of this proposal and therefore not a risk to be mentioned here. To be honest there is no additional storage cost. If the cluster was not hibernated you will anyways take one full snapshot per 24 hrs (default). So when the cluster is hibernated then you only have a cost of single additional copy of the full snapshot at any given point in time. Perhaps it would be nice if you give some cost numbers to back your claim that the cost of one additional full snapshot is minimal. 
Compute the cost assuming the largest possible full snapshot that you might have to maintain an additional copy of.